### PR TITLE
Make cirrus tests presubmit only and add new luci tests.

### PR DESCRIPTION
**Pull request by godofredoc**
## Description

We are now running most of the tests in luci postsubmit and presubmit.
We are removing some of the presubmit tests from cirrus, making some of
the remaining cirruss tests presubmit only and enabling more luci tests
in the flutter dashboard.

## Related Issues

Bug:
  https://github.com/flutter/flutter/issues/63258

## Tests

N/A

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -5,12 +5,12 @@
 # See https://confluence.atlassian.com/bitbucket/yaml-anchors-960154027.html
 
 windows_shard_template: &WINDOWS_SHARD_TEMPLATE
-  only_if: "changesInclude('.cirrus.yml', 'dev/**', 'bin/**') || $CIRRUS_PR == ''" # https://github.com/flutter/flutter/issues/41941
+  only_if: "changesInclude('.cirrus.yml', 'dev/**', 'bin/**') && $CIRRUS_PR == ''" # https://github.com/flutter/flutter/issues/41941
```
Comment (jmagman) [inline .cirrus.yml:8]: Have Windows hostonly been migrated?  I see https://ci.chromium.org/p/flutter/builders/prod/Windows%20hostonly_devicelab_tests but it's failing.

File: .cirrus.yml [hunk 2]
```diff
@@ -5,12 +5,12 @@
...
   script:
     - dart --enable-asserts ./dev/bots/test.dart
 
 macos_shard_template: &MACOS_SHARD_TEMPLATE
-  only_if: "changesInclude('.cirrus.yml', 'dev/**', 'bin/**') || $CIRRUS_PR == ''" # https://github.com/flutter/flutter/issues/41941
+  only_if: "changesInclude('.cirrus.yml', 'dev/**', 'bin/**') && $CIRRUS_PR == ''" # https://github.com/flutter/flutter/issues/41941
```
Comment (jmagman) [inline .cirrus.yml:13]: I don't see hostonly_devicelab_tests for macOS on any builder, am I missing it?

File: .cirrus.yml (hunk 1 above)
Comment (christopherfujino) [inline .cirrus.yml:8]: we want this to ONLY run on post-submit?

File: .cirrus.yml [hunk 3]
```diff
@@ -103,15 +104,8 @@ task:
       script:
         - dart --enable-asserts ./dev/bots/analyze.dart
 
-    - name: fuchsia_precache-linux # linux-only
```
Comment (christopherfujino) [inline .cirrus.yml:106]: So this is moving to LUCI only?

File: .cirrus.yml (hunk 1 above)
Comment (godofredoc) [reply .cirrus.yml]: This was an error, back to ||

File: .cirrus.yml (hunk 2 above)
Comment (godofredoc) [reply .cirrus.yml]: This was an error, back to ||

File: .cirrus.yml (hunk 1 above)
Comment (godofredoc) [reply .cirrus.yml]: Reverted to use ||

**Comment by godofredoc** [comment]
@jmagman @christopherfujino any other comments about this PR?

**Review by jmagman — APPROVED** [review]
Not really familiar with how the builders.json work, but the `.cirrus.yml` LGTM!

**Pull request merged**